### DISCUSSION TITLE
Synchronizing Imported Users with LDAP Users

### DISCUSSION THREAD

**Discussion by duyanhtruong** [kimai/kimai#4991] [Questions & Answers] [answered]
I have Kimai configured for binding wiith LDAP. I started with an empty database, then imported some timesheet entries, which created some users. Then when the user (whose account was created during the imports) tried to log in for the first time, it produces this error:

```[request] CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\UniqueConstraintViolationException: "An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'bruce.wayne@domain.com' for key 'UNIQ_123456789'" at ExceptionConverter.php line 62 {"exception":"[object]```

So it is taking the user from LDAP as a separate user from the existing imported records.

Is there a way I can get the imported user records to synchronize with LDAP users?

**Comment by kevinpapst** [comment]
This has been discussed many times, e.g.: https://github.com/kimai/kimai/discussions/4275#discussioncomment-6907280

**Reply by duyanhtruong** [reply]
I read through that issue too. My apologies if I'm not understanding it right, but I think this one is different. An LDAP user can log in as many times as they want, no problem.

The issue arises when:
- I start with an empty database for an LDAP-configured Kimai
- Import some timesheet entries, which creates new users because there's no pre-existing ones in the database
- When the LDAP user tries to log in for the first time, it gives the duplicate error because their user name and email are already being used for the user that is created from the import.
- So I'm guessing the Import script does not check for users in the LDAP before creating a new user account?

**Reply by kevinpapst** [reply, marked as answer]
No, the importer does not know anything about authentication.
And the issue is always the same: Kimai cannot detect that there is already a username in the database for that DN.

Compare the `kimai2_users` table from the "importer database" and the one only used with LDAP.
 There is an important difference somewhere, likely email or username, depends on your LDAP settings.

**Reply by duyanhtruong** [reply]
You're absolutely right. The imported username was formatted 'First Last' while the LDAP username is 'first.last'. I retested with a new timesheet entry the issue is fixed now.
Thank you for your patience. Your help is much appreciated!
